src/executables/master.c

```diff
@@ -681,20 +681,19 @@ css_send_to_existing_server (CSS_CONN_ENTRY * conn, unsigned short rid, CSS_SERV
 		  if (hb_is_hang_process (temp->fd))
 		    {
 		      css_reject_client_request (conn, rid, SERVER_HANG);
-		      free_and_init (server_name);
 		      css_free_conn (conn);
 		      return;
 		    }
 #endif
 		  if (css_send_new_request_to_server (temp->fd, conn->fd, rid, request))
 		    {
-		      free_and_init (server_name);
 		      css_free_conn (conn);
 		      return;
 		    }
 		  else if (!temp->ha_mode)
 		    {
-		      temp = css_return_entry_of_server (server_name, css_Master_socket_anchor, SERVER_TYPE_ANY);
+		      temp =
+			css_return_entry_of_server (server_name.c_str (), css_Master_socket_anchor, SERVER_TYPE_ANY);
```

Comment:
SERVER_TYPE_ANY or type?

Reply:
should it not be the type delivered by the client via the connection?

Reply:
so type